react/lesson1.md

```diff
@@ -36,175 +91,431 @@ cd my-app/
 npm start
 ```
 
+# Our first component
 
-# JSX
-- Uses [JSX](https://facebook.github.io/react/docs/introducing-jsx.html)
+Components in React can be defined by just declaring a function that return some JSX.
 
-```jsx
-import React, { Component } from 'react';
+- Create a new file in `./src/components/button.js`
 
-class App extends Component {
-  render() {
+- Create your first component
 
-    return (
-      <div className="App">
-        Hello World!
-      </div>
-    );
-  }
+```javascript
+import React from 'react';
+
+function GameButton() {
+    return <button> Awesome button! </button>;
 }
 
-export default App;
+export default GameButton;
 ```
-# Components
+
+- Switch back to `App.js` and use your new custom component!
+
+```javascript
+import GameButton from './components/gameButton';
+// .........
+  <p className="App-intro">
+    <GameButton/>
+  </p>
+// .........
+```
+
+- Check your browser, it should have refreshed with the latest changes
+
+# Class components
 
 React components are best understood in terms of parents and children.
 
 Parent or `container` components may have access to the data that they need in order to pass on to multiple child components. These larger components often contain the actions (e.g. API calls) or data processing functions that their child components require and thus are referred to as `smart` components.
 
-Example:
-```jsx
-import React, { Component } from 'react';
-import Organisation from './Organisation';
+These components are usually defined using ES6 classes. So instead of returning the JSX directly, it's returned as part of a `render()` function. Here's how we would define the button component we created above using ES6 classes:
+
+```javascript
+class GameButton extends React.Components {
 
-class App extends Component {
   render() {
-    return (
-      <div className="App">
-        <Organisation
-            name="Refugee Support Network"
-            website="http://www.refugeesupportnetwork.org"
-        />
-      </div>
-    );
+    return <button> Awesome button! </button>;
   }
 }
+```
+
+The above is the exact same component we defined using the function. 
+
+Child or `presentational` components are essentially smaller customisable to the data that they receive but usually do not have the data in themselves. These are often referred to as `dumb` components - because they do not process the data in themselves:
+
+While function components are a lot simpler to read and write, using ES6 classes allows us to create more powerful, smarter components.
+
+# JSX
+
+You might have noticed something weird: we were returning some HTML-like code in our components.
+
+That is called JSX, and it's a compromise to allows us to use the HTML we are so familiar inside our JavaScript components. 
```

Comment:
_that_ allows

Reply:
I got the term "compromise" from this article: https://medium.freecodecamp.org/all-the-fundamental-react-js-concepts-jammed-into-this-single-medium-article-c83f9b53eac2 (do a CTRL+F for compromise), and I quite liked his explanation.

However, yea, might not be best used in an "into to React" lesson.